packages/lib/src/vis/shared/PanZoomMesh.tsx

```diff
@@ -98,17 +92,18 @@ function PanZoomMesh() {
       const { sourceEvent } = evt;
       const factor = sourceEvent.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
 
-      camera.zoom = Math.max(1, camera.zoom * factor);
+      camera.scale.x = Math.min(1, camera.scale.x / factor);
+      camera.scale.y = Math.min(1, camera.scale.y / factor);
+
       camera.updateProjectionMatrix();
+      camera.updateMatrixWorld();
 
       const projectedPoint = camera.worldToLocal(evt.unprojectedPoint.clone());
-      const { x: pointerX, y: pointerY } = projectedPoint;
-      const { x: camX, y: camY } = camera.position;
-
-      moveCameraTo(
-        camX + pointerX * (1 - 1 / factor),
-        camY + pointerY * (1 - 1 / factor)
+      const scaledPoint = camera.localToWorld(
+        projectedPoint.clone().multiplyScalar(1 - 1 / factor)
       );
```

Comment:
Again, I made the computation more generic by using matrices so that the change in scale is taken into account.

I could simplify this computation to remove projection and unprojection in a future commit.